packages/data/src/countries/resolvers.ts

```diff
@@ -52,6 +53,6 @@ export function* getCountries() {
 
 		return getCountriesSuccess( results );
 	} catch ( error ) {
-		return getCountriesError( error );
+		return getCountriesError( error as RestApiError );
```

Comment:
Hey @chihsuan! Glad to see some work on fixing the Typescript errors!

I always look more carefully when I see type coercion going on - over here we have an unknown error type but we tell TS "as RestApiError", which can be dangerous if the error turns out to not be a RestApiError.

We should probably use a type guard instead, so something like 

```
function isRestApiError(unknownError: unknown) is RestApiError { 
    // check if is RestApiError
    // return true if it is otherwise false
 }
 ```

 And within the catch:
 ``` 
 catch (error) {
      if isRestApiError(error) { return getCountriesError(error) }
      throw error;
 }
 ```

 This way we don't accidentally swallow an unexpected error

Reply:
> I always look more carefully when I see type coercion going on - over here we have an unknown error type but we tell TS "as RestApiError", which can be dangerous if the error turns out to not be a RestApiError.

@rjchow Thanks for bringing this up! I agree with you type coercion can be dangerous and the idea of type guard looks good. 

However, I think we might change the original intention of this function if we throw errors in this catch. From my understanding, we should dispatch the error to`getCountriesError` even it's not an API error. Probably it should be `any` type. Do you have other ideas to improve it?

Reply:
Oh, how about changing the argument type of `getCountriesError` function and other action functions?

```js
export function getCountriesError( error: unknown ) 
```

Reply:
Yea you're right - that would indeed change the intent, I suppose we could check for the necessary keys and if not then do a console.error so that we notice it

Reply:
@rjchow Thanks for the investigation and creating a new issue. https://github.com/woocommerce/woocommerce-admin/issues/8403

Do you think it's still worth adding a check/console.error in this PR?

Reply:
Let's just mark the `data` key as optional in RestApiError for now 🙂

Reply:
Cool. Updated in 72ee87ff57e38a78474d1077d17db469b3ecbdda.